### PR TITLE
feat(project_deatil): Added feature to generate source code bundle

### PR DESCRIPTION
**Pull request by amritkv**
feat(project_detail): Added feature to generate source code bundle at project detail page

Note :  Yet to incorporate with rest-api (as rest-api is under development)

### CODE REVIEW

**Review by heliocastro — CHANGES_REQUESTED** [review]
Code is good, but your editor or prettier config is removing blank last line on the sources.
Please add an extra empty line on end of all this codes and/or adjust your editor.

**Comment by amritkv** [comment]
> Code is good, but your editor or prettier config is removing blank last line on the sources. Please add an extra empty line on end of all this codes and/or adjust your editor.

Sure @heliocastro ! I will do that. 👍

File: src/styles/globals.css [hunk 1]
```diff
@@ -793,4 +794,4 @@ input::-webkit-datetime-edit {
 
 .required {
     color: #F7941E;
-}
```
Comment (heliocastro) [inline src/styles/globals.css:796]: Removing blank line

File: src/app/[locale]/projects/generateSourceCode/[id]/page.tsx
Comment (heliocastro) [inline src/app/[locale]/projects/generateSourceCode/[id]/page.tsx:1]: No blank line in the end of code

File: src/app/[locale]/projects/generateSourceCode/[id]/components/GenerateSourceCodeBundle.tsx
Comment (heliocastro) [inline src/app/[locale]/projects/generateSourceCode/[id]/components/GenerateSourceCodeBundle.tsx:1]: No blank line at end of code

**Review by heliocastro — CHANGES_REQUESTED** [review]
Blank lines

**Review by heliocastro — CHANGES_REQUESTED** [review]
Please fix the log message

File: src/styles/globals.css (hunk 1 above)
Comment (heliocastro) [reply src/styles/globals.css:796]: Still removing a blank line at the end. This time let it pass.

**Review by heliocastro — APPROVED** [review]
Please fix the editor next time

**Comment by heliocastro** [comment]
@amritkv Still WIP ?

**Comment by amritkv** [comment]
> Please fix the editor next time

Hey @heliocastro ! In my case, my `.prettierrc.json` file looks exactly the same as it is in main.
I'm not sure, where the prob is. Can you please help me here.

image

**Comment by amritkv** [comment]
> @amritkv Still WIP ?

I'm not sure whether we can merge it without API integration. If we can, then I can make it ready to merge.

**Comment by heliocastro** [comment]
@amritkv That's why i mentioned your editor. Prettier don't have an configuration for end of line, this comes strictly from the editor.
For example, in Visual Studio Code you need to add this in configuration: files.insertFinalNewline = True
In IDEA IntelliJ is: Settings → Editor → General → Ensure line feed at file end on Save

**Comment by amritkv** [comment]
> iles.insertFinalNewline = True

@heliocastro , The config of my vs code editor is also the same as earlier.
May be I have to update/do a fresh configuration of my editor. 

image

**Pull request merged**